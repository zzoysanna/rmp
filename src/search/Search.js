import React, {Component} from 'react';
import Filter from '../common/Filter';
import './styles/search.less';

export default class Search extends Component {
	constructor(props) {
		super(props);
		this.state = {
			query: '',
			searchBy: 'title',
		};
		this.onInputChange = this.onInputChange.bind(this);
	}

	onInputChange(event) {
		this.setState({
			query: event.target.value,
		});

	}

	onChangeSearchType(type) {
		this.setState({
			searchBy: type,
		});
	}

	onSubmit() {
		console.log(`search "${this.state.query}" by ${this.state.searchBy}`);
	}

	render() {
		const isValidQuery = this.state.query.length >= 3;
		return(
			<div className="search">
				<h1>Find your movie</h1>
				<div className="input-section">
					<input type="text" className="text" value={this.state.query} onChange={this.onInputChange}/>
					<button disabled={!isValidQuery} onClick={() => this.onSubmit()}>Search</button>
				</div>
				<Filter
					title="search by"
					options={['title', 'genre']}
					active={this.state.searchBy}
					onChangeOption={(type) => this.onChangeSearchType(type)}
				/>
			</div>
		);
	}
}
