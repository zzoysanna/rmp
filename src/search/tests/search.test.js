import React from 'react';
import { shallow, mount, render } from 'enzyme';
import Search from "../Search";

describe('Search', () => {

	it('should be render correctly', () => {
		const component = shallow(<Search />);
		expect(component).toMatchSnapshot();
	});
});
