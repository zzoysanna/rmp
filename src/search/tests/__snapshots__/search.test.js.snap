// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search should be render correctly 1`] = `ReactWrapper {}`;

exports[`Search should be render correctly on button click 1`] = `
Array [
  Object {
    "type": "CHANGE_SEARCH_TYPE",
  },
]
`;
